### PR TITLE
Add plybooks for EAP 7.2 and include JCLIFF

### PR DESCRIPTION
**Pull request by hcherukuri**

### CODE REVIEW

**Review by sabre1041 — CHANGES_REQUESTED** [review]
Seeing the following errors:

1. On first run:

```
TASK [jcliff] ******************************************************************************************************************************************************************************************
Wednesday 05 August 2020  14:51:32 -0500 (0:00:01.618)       0:25:25.311 ****** 
fatal: [10.9.51.52]: FAILED! => {"changed": false, "msg": "wfly_home is invalid: /opt/jboss-as/jboss-eap-7.2/ "}
```

2.  Errors on subsequent runs

```
TASK [redhat-cop.jboss_eap : Async Download JBoss EAP Patch from Red Hat Customer Portal] **************************************************************************************************************
Wednesday 05 August 2020  14:58:13 -0500 (0:00:00.432)       0:05:18.308 ****** 
fatal: [10.9.51.52]: FAILED! => {"msg": "The conditional check 'transfer_method == 'csp-to-host' and jboss_eap_apply_patch | bool and (not jboss_eap_exists.stat.exists | bool or 'base' in patch_info_result.stdout)' failed. The error was: error while evaluating conditional (transfer_method == 'csp-to-host' and jboss_eap_apply_patch | bool and (not jboss_eap_exists.stat.exists | bool or 'base' in patch_info_result.stdout)): Unable to look up a name or access an attribute in template string ({% if transfer_method == 'csp-to-host' and jboss_eap_apply_patch | bool and (not jboss_eap_exists.stat.exists | bool or 'base' in patch_info_result.stdout) %} True {% else %} False {% endif %}).\nMake sure your variable name does not contain invalid characters like '-': argument of type 'AnsibleUndefined' is not iterable\n\nThe error appears to be in '/Users/ablock/Projects/RHC/internal/cop/git/ansible-middleware/redhat-cop.jboss_eap/tasks/apply_patch.yml': line 43, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- name: Async Download JBoss EAP Patch from Red Hat Customer Portal\n  ^ here\n"}
```

**Comment by hcherukuri** [comment]
The problem is the first two role here are redhat-cop.jboss_common and redhat-cop.jboss_eap are two different projects and I am getting errors with Ansible 2.9. If they execute well then we have jboss-eap-7.2 on the VM then we can run the jcliff role. I don't see any where we are configuring JBOSS_HOME in any of those roles. So I have checked where it is been held and passed jboss_home for jcliff.

I tried jcliff code it works without any issues, I will work on trouble shooting the first two roles.

**Comment by hcherukuri** [comment]
As mentioned to you on chat, https://github.com/redhat-cop/jboss_eap/blob/master/defaults/main.yml includes config of 7.1 so files generated will be under /opt/jboss-as/jboss-eap-7.1/, so we are seeing jboss_home issue.

**Comment by sabre1041** [comment]
@hcherukuri No longer receiving the first error as described above. Am still facing the second error

**Comment by hcherukuri** [comment]
I don't see that error while running, could you let me know the command and parameters you are passing for running the 7.2 playbook. Looks to me as syntax issue. Could you apply these changes and try, https://github.com/redhat-cop/jboss_eap/pull/65.

and on which build are you running the playbook on ?

File: requirements.yml [hunk 1]
```diff
@@ -1,25 +1,30 @@
...
 
-- src: redhat-cop.redhat_sso
-  name: redhat-cop.redhat_sso
+  - src: sabre1041.redhat-csp-download
+    name: sabre1041.redhat-csp-download
 
-- src: sabre1041.redhat-csp-download
-  name: sabre1041.redhat-csp-download
+
+collections:
```
Comment (sabre1041) [inline requirements.yml:28]: Can we align this with the new `wildfly.jcliff` collection? This is updated location

File: eap7.2-rhel7.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  vars:
+    app_url: http://people.redhat.com/~rpelisse/simple-webapp.war
+    app: /tmp/simple-webapp.war
+    ansible_distribution: RedHat
+    ansible_os_family: RedHat
+    ansible_pkg_mgr: yum
+    jboss_home: /opt/jboss-as/jboss-eap-7.1/
+
+  collections:
+    - redhat.jcliff
```
Comment (sabre1041) [inline eap7.2-rhel7.yml]: Update to wildfly.jcliff

**Review by sabre1041 — CHANGES_REQUESTED** [review]
Everything else checks out. Please just update the collection name/version and we can merge it in!

File: eap7.2-rhel7.yml (hunk 2 above)
Comment (hcherukuri) [reply eap7.2-rhel7.yml]: Updated.

**Comment by hcherukuri** [comment]
Updated wildfly.jcliff in the requirments.yml and playbooks. Please proceed.

File: requirements.yml [hunk 3]
```diff
@@ -1,25 +1,30 @@
...
+  - name: wildfly.jcliff
+    verison: 0.0.1
```
Comment (sabre1041) [inline requirements.yml]: 0.0.2 is the version we need to use

**Review by sabre1041 — CHANGES_REQUESTED** [review]

**Review by sabre1041 — APPROVED** [review]
LGTM

**Pull request merged**